server/routes/index.ts

```diff
@@ -1,136 +1,43 @@
-    printout = JSON.stringify(s)
-  } else {
-    printout = s
-  }
-  if (printout.length > len) {
-    printout = printout.slice(0, len) + '...' + printout.slice(printout.length - 5)
-  }
-  return printout
-}
-const cached = (ttl?: number) => (req: Request, res: Response, next: NextFunction): void => {
-  const key = `${req.method}|${req.path}|${JSON.stringify(req.query)}|${JSON.stringify(req.body)}`
-  const keyContentType = key + '|header|content-type'
-  const v = cache.get(key)
-  if (v) {
-    const contentType = cache.get(keyContentType)
-    if (contentType) {
-      console.log(`Cache header hit key=[${keyContentType}] value=`, contentType)
-      res.header('content-type', contentType as string)
-    }
-    console.log(`Cache hit key=[${key}] value=`, abbrv(v), typeof v)
-    res.send(v)
-    return
-  } else {
-    // @ts-expect-error wrapper
-    res.__send = res.send
-    res.send = (r) => {
-      console.log(`Cache set key=[${key}] value=`, abbrv(r), typeof r)
-      cache.set(key, r, { ttl: ttl ?? 60 * 1000 })
-      if (res.hasHeader('content-type')) {
-        const h = res.getHeader('content-type')
-        console.log(`Cache header set key=[${keyContentType}] value=`, h)
-        cache.set(keyContentType, h, { ttl: ttl ?? 60 * 1000 })
-      }
-      // @ts-expect-error wrapper
-      return res.__send(r)
-    }
-  }
-  next()
-}
 
 router.get('/health', async (req, res) => {
   console.log('[/health]', JSON.stringify(req.fingerprint))
   res.send('OK').end()
 })
 
-const substackDomainCache = {
-  // uncomment the below for test purpose
-  'localhost:3100': 'polymorpher.substack.com'
-}
-
-const client = buildClient()
-
 router.get('/substack/api/v1/archive',
   limiter(),
+  substackDomainCache,
   async (req, res) => {
-    const host = req.get('host')
-
-    if (!host) {
-      return
-    }
-
-    let substackDomain = substackDomainCache[host]
-
-    if (!substackDomain) {
-      const [subdomain, sld] = host?.split('.')
-
-      substackDomain = await client.getLandingPage(sld, subdomain)
-
-      if (!substackDomain.endsWith('.substack.com')) {
-        throw new Error('Not substack page')
-      }
-
-      substackDomainCache[host] = substackDomain
-    }
-
+    const { substackDomain } = res.locals
     const { data } = await axiosBase.get(`https://${substackDomain}/api/v1/archive`, { params: req.query })
 
-    res.status(200).send(data)
+    res.status(200).set('content-type', 'application/json; charset=utf-8').send(data)
   }
 )
 
 const axiosBase = axios.create({ timeout: 15000 })
 
 router.get('/substack',
   limiter(),
+  substackDomainCache,
   async (req, res) => {
     try {
-      if (!req.query.url) {
-        throw new Error('URL query param is not specified')
-      }
-
-      const url = decodeURI(req.query.url as string)
-
-      if (!isValidSubstackUrl(url)) {
-        throw new Error('Not substack url')
-      }
+      const url = decodeURI(req.query.url as string ?? '')
```

Comment:
I thought express handles decoding URI automatically, does it not?

Reply:
Removed `decodeURI()`